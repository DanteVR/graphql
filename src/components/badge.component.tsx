import React from 'react';
import { StyleSheet, Text, View, ViewProps } from 'react-native';
import { pallette, utilities } from '../utils/constants';

interface BadgeProps extends ViewProps {
  type?: 'primary' | 'secondary';
  text: string;
}

export const Badge = ({ type = 'primary', text, ...next }: BadgeProps) => {
  return (
    <View {...next} style={[styles.badge, styles[type]]}>
      <Text style={styles.label}>{text}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  badge: {
    padding: utilities.padding.xs,
    borderRadius: utilities.borderRadius,
    alignSelf: 'flex-start',
    backgroundColor: pallette.primary,
    marginBottom: utilities.margin.md,
  },
  label: {
    color: '#fff',
  },
  primary: {
    backgroundColor: pallette.primary,
  },
  secondary: {
    backgroundColor: pallette.secondary,
  },
});
