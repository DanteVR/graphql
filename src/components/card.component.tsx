import React from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  TouchableOpacityProps,
} from 'react-native';

export interface CardProps extends TouchableOpacityProps {}

export const Card = ({ children, style, ...next }: CardProps) => {
  return (
    <TouchableOpacity
      style={[styles.card, style]}
      activeOpacity={0.8}
      {...next}>
      {children}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  card: {
    flex: 1,
    backgroundColor: '#000',
  },
});
