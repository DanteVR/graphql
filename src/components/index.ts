import { Loading } from './loading.component';
import { Card } from './card.component';
import { StatusBar } from './statusbar.component';
import { Badge } from './badge.component';

export { Loading, Card, StatusBar, Badge };
