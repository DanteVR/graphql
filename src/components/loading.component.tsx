import React from 'react';
import {
  ActivityIndicator,
  ActivityIndicatorProps,
  StyleSheet,
  View,
} from 'react-native';
import { pallette } from '../utils/constants';

interface LoadingProps extends ActivityIndicatorProps {}

export const Loading = ({
  size = 'large',
  color = pallette.primary,
  ...next
}: LoadingProps) => {
  return (
    <View style={styles.container}>
      <ActivityIndicator size={size} color={color} {...next} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: '#fff',
  },
});
