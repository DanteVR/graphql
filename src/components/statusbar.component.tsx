import React from 'react';
import { StatusBar as RNStatusBar, StatusBarProps } from 'react-native';
import { pallette } from '../utils/constants';

export const StatusBar = (props: StatusBarProps) => {
  return <RNStatusBar {...props} backgroundColor={pallette.primary} />;
};
