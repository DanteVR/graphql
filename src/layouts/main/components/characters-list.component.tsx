import React, { useCallback } from 'react';
import { FlatList } from 'react-native';
import { Item } from '../components/item.component';

interface ListProps {
  data: any;
}

export const CharactersList = ({ data }: ListProps) => {
  const renderItem = useCallback(
    ({ item: { image } }: any) => <Item image={image} />,
    [],
  );

  return (
    <FlatList
      numColumns={2}
      data={data}
      renderItem={renderItem}
      keyExtractor={character => character.id.toString()}
    />
  );
};
