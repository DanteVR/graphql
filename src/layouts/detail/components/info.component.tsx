import React from 'react';
import { Image, StyleSheet, View } from 'react-native';
import { Character } from '../../../schemas/character.schema';
import { Badge } from '../../../components';
import { utilities } from '../../../utils/constants';

interface DetailProps {
  character: Character;
}

export const Info = ({ character }: DetailProps) => {
  return (
    <View style={styles.container}>
      <Image
        style={styles.image}
        source={{
          uri: character.image,
        }}
      />
      <View style={styles.info}>
        <Badge text={`Name: ${character.name}`} />
        <Badge text={`Species: ${character.species}`} type="secondary" />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  image: {
    flex: 1,
    resizeMode: 'cover',
  },
  info: {
    position: 'absolute',
    top: utilities.margin.md,
    left: utilities.margin.md,
  },
});
