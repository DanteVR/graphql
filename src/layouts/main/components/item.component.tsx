import React from 'react';
import { Card } from '../../../components';
import { Image, StyleSheet } from 'react-native';
import { screen } from '../../../utils/constants';
import { MainScreenNavigationProp } from '..';
import { useNavigation } from '@react-navigation/native';
import { Character } from '../../../schemas/character.schema';

interface ItemProps {
  character: Character;
}

export const Item = ({ character }: ItemProps) => {
  const navigation = useNavigation<MainScreenNavigationProp>();

  return (
    <Card onPress={() => navigation.navigate('Detail', { ...character })}>
      <Image
        style={styles.image}
        source={{
          uri: character.image,
        }}
      />
    </Card>
  );
};

const styles = StyleSheet.create({
  image: {
    width: screen.width / 2,
    height: screen.width / 2,
    resizeMode: 'cover',
  },
});
