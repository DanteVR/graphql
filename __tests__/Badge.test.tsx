import 'react-native';
import React from 'react';
import render from 'react-test-renderer';
import { it, expect } from '@jest/globals';

import { Badge } from '../src/components';

it('Renders correctly with the given text', () => {
  const tree = render.create(<Badge text="This is a text" />).toJSON();
  expect(tree).toMatchSnapshot();
});
