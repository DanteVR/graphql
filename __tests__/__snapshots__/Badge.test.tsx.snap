// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with the given text 1`] = `
<View
  style={
    [
      {
        "alignSelf": "flex-start",
        "backgroundColor": "#081B33",
        "borderRadius": 4,
        "marginBottom": 12,
        "padding": 8,
      },
      {
        "backgroundColor": "#081B33",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      {
        "color": "#fff",
      }
    }
  >
    This is a text
  </Text>
</View>
`;
